src/relax/transform/static_plan_block_memory.cc

```diff
@@ -476,11 +521,15 @@ class StorageAllocatorInit : public StorageAllocatorBaseVisitor {
     token2block_.erase(token_to_discard.get());
   }
 
+  /*! \brief The arithmetic analyzer. */
+  arith::Analyzer ana;
   /*!
    * \brief The context IRModule, used for checking if a callee function is
    * a PrimFunc inside the IRModule.
    */
   const IRModule& ctx_mod_;
+  /*! \brief The mapping from TIR variables to their respective upper bound values. */
+  std::unordered_map<const tir::VarNode*, IntImm> var_upper_bound_;
```

Comment:
prefer `unordered_map<tir::Var, IntImm, ObjectPtrHash, ObjectPtrEqual>`, mainly to avoid var get de-allocated and re-allocated, does not matter in here